spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerScanner.java

```diff
@@ -114,9 +115,9 @@ public InputPartition[] planInputPartitions() {
               .collect(Collectors.toList());
 
       return parts.toArray(new InputPartition[0]);
-    } catch (Exception e) {
-      log.error("planInputPartitions exception: " + e);
-      return null;
+    } catch (JsonProcessingException e) {
+      throw new SpannerConnectorException(
+          SpannerErrorCode.SPANNER_FAILED_TO_PARSE_OPTIONS, "Error parsing the input options.", e);
```

Comment:
The reason we didn't throw an exception here is because this method to be overridden doesn't allow for exceptions but perhaps the callers can handle the exception.

Reply:
We wrapped the SpannerConnectorException extending the RuntimeException, which doesn't affect the interface. The runtimeException will be thrown to the customer directly if it's called.

Reply:
Gotcha and thank you! LGTM!